### PR TITLE
Size computation mode

### PR DESCRIPTION
**Pull request by xokker**
This PR is based on https://github.com/google/data-transfer-project/pull/1079. Please see the second commit of this PR (3d6b9fbe4d42742406314bfd254eaa7193e817e7) for changes related to the size computation mode.

### CODE REVIEW

**Comment by wmorland** [comment]
Investigating VideosContainerResourceTest failure

File: portability-transfer/src/main/java/org/datatransferproject/transfer/CallableSizeCalculator.java
Comment (jzacsh) [inline portability-transfer/src/main/java/org/datatransferproject/transfer/CallableSizeCalculator.java:32]: (shouldn't block PR) consideration for the future: If I'm understanding this right, then I think we could use a [counting proxy](https://commons.apache.org/proper/commons-io/javadocs/api-2.5/org/apache/commons/io/input/CountingInputStream.html) and then maybe just have a single pass

File: portability-transfer/src/main/java/org/datatransferproject/transfer/copier/PortabilityAbstractInMemoryDataCopier.java
Comment (jzacsh) [inline portability-transfer/src/main/java/org/datatransferproject/transfer/copier/PortabilityAbstractInMemoryDataCopier.java:232]: (shouldn't block PR) another future consideration is make `getItems` an interface all models are required to have so we don't have to maintain this coverage. We should definitely at least come back to this code after you merge and drop a TODO here to fill out the coverage (either via interface or by hand for the current known models).

File: portability-types-common/src/main/java/org/datatransferproject/types/common/models/videos/VideoModel.java
Comment (jzacsh) [inline portability-types-common/src/main/java/org/datatransferproject/types/common/models/videos/VideoModel.java:26]: (maybe shouldn't block this PR - but also I'd be surprised if this compiles) I think this might be missing `ImportableItem` requirement of `getIdempotentId` maybe? Anyway if it compiles, feel free to revisit later.

File: portability-spi-transfer/src/main/java/org/datatransferproject/spi/transfer/idempotentexecutor/ItemImportResult.java
Comment (jzacsh) [inline portability-spi-transfer/src/main/java/org/datatransferproject/spi/transfer/idempotentexecutor/ItemImportResult.java:1]: (same) skipping review as I think this belongs to #1079 (feel free to merge in either - but I only reviewed the other copy)

File: portability-spi-transfer/src/main/java/org/datatransferproject/spi/transfer/idempotentexecutor/IdempotentImportExecutor.java
Comment (jzacsh) [inline portability-spi-transfer/src/main/java/org/datatransferproject/spi/transfer/idempotentexecutor/IdempotentImportExecutor.java:16]: (same) skipping review as I think this belongs to #1079 (feel free to merge in either - but I only reviewed the other copy)

File: portability-spi-transfer/src/main/java/org/datatransferproject/spi/transfer/idempotentexecutor/CachingExecutor.java
Comment (jzacsh) [inline portability-spi-transfer/src/main/java/org/datatransferproject/spi/transfer/idempotentexecutor/CachingExecutor.java:1]: skipping review as I think this belongs to #1079 (feel free to merge in either - but I only reviewed the other copy)

File: portability-types-common/src/main/java/org/datatransferproject/types/common/models/photos/PhotoModel.java
Comment (jzacsh) [inline portability-types-common/src/main/java/org/datatransferproject/types/common/models/photos/PhotoModel.java:22]: (same) skipping review as I think this belongs to #1079 (feel free to merge in either - but I only reviewed the other copy)

**Review by jzacsh — DISMISSED** [review]

**Comment by jzacsh** [comment]
[as with PR #1079](https://github.com/google/data-transfer-project/pull/1079#issuecomment-1159267663) feel free to merge - just left comments for your consideration later. The one compilation comment I made is probably moot since clearly the bot ran some gradle stuff so I guess we're all good.

File: portability-types-common/src/main/java/org/datatransferproject/types/common/models/videos/VideoModel.java
Comment (wmorland) [reply portability-types-common/src/main/java/org/datatransferproject/types/common/models/videos/VideoModel.java:26]: It inherits the Interface from the slightly peculiar schema.org style Thing.java

**Review by wmorland — APPROVED** [review]
Merged changes

**Pull request merged**
